File: app/api/schemas/product.py
from pydantic import BaseModel, Field
import uuid


class CategoryBase(BaseModel):
    name: str


class CategoryCreate(CategoryBase):
    pass


class CategoryOut(CategoryBase):
    id: uuid.UUID
    name: str

    model_config = {"from_attributes": True}


## Product


class ProductBase(BaseModel):
    title: str
    description: str | None = None
    price: float = Field(gt=0)
    stock_quantity: int = Field(ge=0)


class ProductCreate(ProductBase):
    category_ids: list[uuid.UUID] | None = None


class ProductOut(ProductBase):
    id: uuid.UUID
    title: str
    description: str
    slug: str
    price: float
    categories: list[CategoryOut] = Field(default_factory=list)
    image_url: str | None = None

    model_config = {
        "from_attributes": True,
    }


class PaginatedProductOut(BaseModel):
    total: int
    page: int
    limit: int
    items: list[ProductOut]

File: frontend/src/lib/types/product.ts
import { Category } from "./category";

export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  stock_quantity: number;
  slug: string;
  categories: Category[];
  image_url: string;
  created_at: string;
  updated_at: string;
}

export interface ProductApiResponse {
  total: number;
  page: number;
  limit: number;
  items: Product[];
}
